pkg/cloudprovider/gcp/block_store.go

```diff
@@ -212,7 +214,16 @@ func getSnapshotTags(arkTags map[string]string, diskDescription string, log logr
 func (b *blockStore) DeleteSnapshot(snapshotID string) error {
 	_, err := b.gce.Snapshots.Delete(b.project, snapshotID).Do()
 
-	return errors.WithStack(err)
+	// if it's a 404 (not found) error, we don't need to return an error
+	// since the snapshot is not there.
+	if gcpErr, ok := err.(*googleapi.Error); ok && gcpErr.Code == http.StatusNotFound {
+		return nil
+	}
+	if err != nil {
```

Comment:
This is super nit, might want to ignore: it'd read better if the flow of handling error cases here was err, err nil, as opposed to nil, err, nil.

Reply:
OK, not going to address on this PR since this is a backport of existing master commits, but we can discuss separately :)

Reply:
Fair enough!